config/initializers/blacklight_helpers.rb

```diff
@@ -24,6 +24,6 @@ def link_to_document(doc, field_or_opts = nil, opts = { counter: nil })
   # @param [Object] SOLR document
   # @return [String] url string
   def add_highlight_url(doc)
-    params.key?(:q) ? url_for_document(doc).to_s + "/highlight/" + CGI.escape(params['q']) : url_for_document(doc)
+    params.key?(:q) && !params['q'].empty? ? url_for_document(doc).to_s + "/highlight/" + CGI.escape(params['q']) : url_for_document(doc)
```

Comment:
You might change `!empty?` to `present?` -- I _think_ that's the preferred Hyrax/Ruby way?  (although it is one extra character...)

Reply:
That works for both checks so removes the need for &&. Thanks!